contracts/src/instances/ezeth/ezETHBase.sol

```diff
@@ -0,0 +1,156 @@
+        unchecked {
+            refund = msg.value - _baseAmount;
+        }
+
+        uint256 balanceBefore = _ezETH.balanceOf(address(this));
+        // Submit the provided ether to Renzo to be deposited.
+        // a referrer id can be put in here
+        _restakeManager.depositETH{ value: _baseAmount }();
+        uint256 balanceAfter = _ezETH.balanceOf(address(this));
+        sharesMinted = balanceAfter - balanceBefore;
+
+        return (sharesMinted, refund);
+    }
+
+    /// @dev Process a deposit in vault shares.
+    /// @param _shareAmount The vault shares amount to deposit.
+    function _depositWithShares(
+        uint256 _shareAmount,
+        bytes calldata // unused
+    ) internal override {
+        // Transfer ezETH shares into the contract.
+        _ezETH.transferFrom(msg.sender, address(this), _shareAmount);
+    }
+
+    /// @dev Process a withdrawal in base and send the proceeds to the
+    ///      destination.
+    function _withdrawWithBase(
+        uint256, // unused
+        address, // unused
+        bytes calldata // unused
+    ) internal pure override returns (uint256) {
+        // ezETH withdrawals aren't necessarily instantaneous. Users that want
+        // to withdraw can manage their withdrawal separately.
+        revert IHyperdrive.UnsupportedToken();
+    }
+
+    /// @dev Process a withdrawal in vault shares and send the proceeds to the
+    ///      destination.
+    /// @param _shareAmount The amount of vault shares to withdraw.
+    /// @param _destination The destination of the withdrawal.
+    function _withdrawWithShares(
+        uint256 _shareAmount,
+        address _destination,
+        bytes calldata // unused
+    ) internal override {
+        // Transfer the ezETH shares to the destination.
+        _ezETH.transfer(_destination, _shareAmount);
+    }
+
+    /// @dev Convert an amount of vault shares to an amount of base.
+    /// @param _shareAmount The vault shares amount.
+    /// @return baseAmount The base amount.
+    function _convertToBase(
+        uint256 _shareAmount
+    ) internal view override returns (uint256) {
+        (, , uint256 totalTVL) = _restakeManager.calculateTVLs();
+        uint256 ezETHSupply = _ezETH.totalSupply();
+
+        // Price in ETH / ezETH, does not include eigenlayer points.
+        uint256 sharePrice = totalTVL.divDown(ezETHSupply);
+
+        return _shareAmount.mulDown(sharePrice);
+    }
+
+    /// @dev Convert an amount of base to an amount of vault shares.
+    /// @param _baseAmount The base amount.
+    /// @return shareAmount The vault shares amount.
+    function _convertToShares(
+        uint256 _baseAmount
+    ) internal view override returns (uint256) {
+        (, , uint256 totalTVL) = _restakeManager.calculateTVLs();
+        uint256 ezETHSupply = _ezETH.totalSupply();
+
+        // Price in ETH / ezETH, does not include eigenlayer points.
+        uint256 sharePrice = totalTVL.divDown(ezETHSupply);
```

Comment:
This is one of the culprits for the loose error bounds. We should be using their function: https://github.com/Renzo-Protocol/contracts-public/blob/0530cbd20590677da5fa6fd32a0a8b250a9eb028/contracts/Oracle/RenzoOracle.sol#L111.